### DISCUSSION TITLE
ETA for Stable Version (V1)

### DISCUSSION THREAD

**Discussion by JonathanCauchi** [Snowflake-Labs/terraform-provider-snowflake#2958] [General] [answered]
Hi all,

I work as an analytics engineer at a company where I've been working on Snowflake access control with our DevOps team. We identified this provider as a tool but we're waiting for a stable version (V1) to be released first before using it. Any ETA for V1 or any stable release? Also if there could be some way to communicate directly once a stable version is out, that would be great.

Best regards,
Jonathan

**Comment by sfc-gh-jcieslak** [comment]
Hey 👋 
Here's the [roadmap](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/ROADMAP.md) that may clear things up. Right now, we're focusing on stabilizing the provider on different levels and providing new features that are crucial for V1. There are also additional documents ([here](https://github.com/Snowflake-Labs/terraform-provider-snowflake/tree/main/v1-preparations)) that we update to provide a wider view on V1 progress and design decisions we had to take (we're doing more than creating/refactoring resources, but it's the biggest task and a pretty good indicator of V1 readiness I think). With every quarter we update the roadmap to communicate where we are at and what are our plans. I cannot give any ETA for now, but the V1 release will be for sure announced in GitHub discussions (and maybe other mediums).

**Reply by sfc-gh-jcieslak** [reply, marked as answer]
It also depends on what you will be trying to manage with our provider. For example, grants are already refactored and there's a small probability we'll touch them before V1. Other resources, like [database](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/database) are also already refactored, but they're marked as V1-candidate, meaning it may still change, but all the breaking changes are documented in the [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md) that helps with upgrading.
